registration-processor/mosip-stage-executor/src/main/java/io/mosip/registration/processor/stages/executor/util/StageClassesUtil.java

```
@@ -0,0 +1,137 @@
+	 * @return the stage bean classes
+	 */
+	@SuppressWarnings("unchecked")
+	public static List<Class<MosipVerticleAPIManager>> getStageBeanClasses(List<String> stageBeansBasePackages) {
+		Set<String> stageClassNamesSet = getStageBeanClassNames(stageBeansBasePackages);
+		
+		List<Class<MosipVerticleAPIManager>> stageClasses = stageClassNamesSet.stream()
+				.map(classStr -> {
+					try {
+						return (Class<MosipVerticleAPIManager>) Class.forName(classStr);
+					} catch (ClassNotFoundException e1) {
+						regProcLogger.error("Unable to load stage class : \n{}", ExceptionUtils.getStackTrace(e1));
+						throw new RuntimeException("Invalid config", e1);
+					}
+				}).filter(Objects::nonNull).collect(Collectors.toList());
+		return stageClasses;
+	}
+
+	/**
+	 * Gets the stage bean class names.
+	 *
+	 * @param stageBeansBasePackages the stage beans base packages
+	 * @return the stage bean class names
+	 */
+	public static Set<String> getStageBeanClassNames(List<String> stageBeansBasePackages) {
+		Set<BeanDefinition> stagesBeanDefinitions = getStagesBeanDefinitions(stageBeansBasePackages);
+		
+		Set<String> stageClassesSet = stagesBeanDefinitions.stream().map(bd -> bd.getBeanClassName())
+											.collect(Collectors.toSet());
+		return stageClassesSet;
+	}
+
+	/**
+	 * Gets the stage beans base packages.
+	 *
+	 * @param stagesConfig the stages config
+	 * @param propertySources the property sources
+	 * @return the stage beans base packages
+	 */
+	public static List<String> getStageBeansBasePackages(StagesConfig stagesConfig, MutablePropertySources propertySources) {
+		String stageBeansBasePkgsPropertyName= PROP_STAGE_GROUPS_STAGE_BEANS_BASE_PACKAGES_PREFIX + stagesConfig.getStageGroupName();
+		PropertySourcesPropertyResolver propertySourcesPropertyResolver = new PropertySourcesPropertyResolver(propertySources);
+		String defaultStageBeanBasePkgStr = propertySourcesPropertyResolver.getProperty(PROP_DEFAULT_STAGE_GROUP_STAGE_BEANS_BASE_PACKAGE, 

```

Can we remove this default value, if property is not available, lets throw the error and stop. Too many defaults is confusing.

---

Done.